src/core/src/main/java/org/apache/jmeter/testelement/TestPlan.java

```diff
@@ -196,11 +226,22 @@ public String getTestPlanClasspath() {
 
     /**
      * Fetch the serialize threadgroups property
-     *
+     * Could be change for no gui test with jmeter property : PROP_SERIALIZE_THREADGROUPS
      * @return serialized setting
      */
     public boolean isSerialized() {
-        return getPropertyAsBoolean(SERIALIZE_THREADGROUPS);
+        boolean bSerializedThreadGroupDefault = getPropertyAsBoolean(SERIALIZE_THREADGROUPS);
```

Comment:
This seems to be the third replication of the logic to override a value by another value, if non-gui mode is detected and log those decisions. Might be replaced by a general (private) method.